### PR TITLE
Better non-root error

### PR DESCRIPTION
**Pull request by williangaspar**

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -54,6 +55,12 @@ static void enforce_infinite_rlimit() {
 
 int main(int argc, char *argv[])
 {
+  if (geteuid() != 0)
+  {
+    std::cerr << "ERROR: bpftrace currently only supports running as the root user." << std::endl;
+    exit(0);
```
Comment (mmarchini) [inline src/main.cpp]: Change exit status code to non-zero to indicate bpftrace exited with error: 

```suggestion
    exit(1);
```

**Comment by danobi** [comment]
Can this check be moved to after argument parsing? The patch right now would require root to print the help text.

File: src/main.cpp [hunk 2]
```diff
@@ -52,8 +53,18 @@ static void enforce_infinite_rlimit() {
...
+  if (geteuid() != 0)
+  {
+    std::cerr << "ERROR: bpftrace currently only supports running as the root user." << std::endl;
+    exit(1);
+  }
+}
+
 int main(int argc, char *argv[])
 {
+
```
Comment (danobi) [inline src/main.cpp]: Extra whitespace

**Comment by brendangregg** [comment]
Thanks!

**Pull request merged**